specification/ml/_types/TrainedModel.ts

```diff
@@ -31,39 +31,114 @@ import {
+  /**
+   * The number of times the model was loaded for inference and was not retrieved from the cache.
+   * If this number is close to the `inference_count`, the cache is not being appropriately used.
+   * This can be solved by increasing the cache size or its time-to-live (TTL).
+   * Refer to general machine learning settings for the appropriate settings.
+   * @doc_id ml-settings
+   */
+  cache_miss_count: integer
   /** The number of failures when using the model for inference. */
-  failure_count: long
-  /** The total number of times the model has been called for inference. This is across all inference contexts, including all pipelines. */
-  inference_count: long
+  failure_count: integer
   /**
-   * The number of times the model was loaded for inference and was not retrieved from the cache. If this number is close to the inference_count, then the cache is not being appropriately used. This can be solved by increasing the cache size or its time-to-live (TTL). See General machine learning settings for the appropriate settings.
-   * @doc_url https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-settings.html#general-ml-settings
+   * The total number of times the model has been called for inference.
+   * This is across all inference contexts, including all pipelines.
    */
-  cache_miss_count: long
+  inference_count: integer
   /** The number of inference calls where all the training features for the model were missing. */
-  missing_all_fields_count: long
+  missing_all_fields_count: integer
   /** The time when the statistics were last updated. */
   timestamp: Time
 }
 
+export class TrainedModelSizeStats {
+  /** The size of the model in bytes. */
+  model_size_bytes: ByteSize
+  /** The amount of memory required to load the model in bytes. */
+  required_native_memory_bytes: integer
+}
+
+export class TrainedModelDeploymentNodesStats {
+  /** The average time for each inference call to complete on this node. */
+  average_inference_time_ms: double
+  /** The number of errors when evaluating the trained model. */
+  error_count: integer
+  /** The total number of inference calls made against this node for this model. */
+  inference_count: integer
+  /**
+   * The number of threads used by the inference process. This value is limited
+   * by the number of hardware threads on the node; it might therefore differ
+   * from the `inference_threads` value in the start trained model deployment API.
+   */
+  inference_threads: integer
+  /** The epoch time stamp of the last inference call for the model on this node. */
+  last_access: long
+  /**
+   * The number of threads used when sending inference requests to the model.
+   * This value is limited by the number of hardware threads on the node; it
+   * might therefore differ from the `model_threads` value in the start trained
+   * model deployment API.
+   */
+  model_threads: integer
+  /** Information pertaining to the node. */
+  node: DiscoveryNode
```

Comment:
One more field:
```
/** The number of inference requests queued to be processed */
number_of_pending_requests: integer 
```

Reply:
Thanks, I've added that field in 9f90cf4dcf266a3bdbb96f10a0f24f5688c6d76b